src/models/oidc/Providers.ts

```diff
@@ -0,0 +1,31 @@
+export interface BaseOidcProvider {
```

Comment:
I also feel that the `types` file referencing all service interfaces and all models is getting cumbersome. Here I have started the implementation as if were a traditional service.

Alternative I can put it all in the `types` file and create a reference repository ticket to change that at a later date.